### ISSUE TITLE
Support for Regex Validation of Text Inputs

### ISSUE THREAD

**Issue by saladd-bu** [bartreardon/swiftDialog#129]
When using Dialog to gather information via textfields, it would be helpful to have the option to validate a user's input at the time they submit the form. This can improve accuracy by only recording data provided in an expected format (for example, in scenarios where a provided input should align to a standard). This can also reduce frustration by avoiding the need to re-display the form should a provided value not be formatted as expected, allowing the user to see the error prior to the window closing and allowing them to edit what they provided and try again.

This might be accomplished by checked the provided value against a regex string at the time the user clicks the default button. If validation is successful, Dialog outputs the values and exits 0.

If validation fails, Dialog would follow a similar functionality as when a required field is left blank. The window would not close and the field would be highlighted.

Suggesting two optional properties be added to textfield for this:
`validation` : Accepts regex patterns
`validationerror` : (Nice to have) Accepts a string so an admin may define custom error text to display above or below the field that failed validation. If not provided, could default to something like "{Textfield Name} is not in the correct format."

Example (validating if technician provided Asset Tag value is a six digit string:

```dialog --textfield "Asset Tag",required,prompt="000000",validation='^\d{6}$',validationerror="The asset tag provided is not valid."```

```
{
  "textfield" : [
    {"title" : "Asset Tag",
     "required" : true,
     "prompt" : "000000",
     "validation" : "^\\d{6}$",
     "validationerror" : "The asset tag provided is not valid."}
  ]
}
```

**Comment by bartreardon** [comment]
@saladd-bu let me know what you think of this:

![image](https://user-images.githubusercontent.com/3598965/170858447-5269fd56-c087-4466-bda0-e5fba14c2be3.png)

I haven't settled on the name of the fields yet. ("validation" or "regex" etc)

**Comment by bartreardon** [comment]
Addressed in https://github.com/bartreardon/swiftDialog/commit/9c098796b9638aaa570d508656a90aff419a53b4 

Merged into dev branch and will tweak a little but this will be a feature release in v1.11

**Comment by saladd-bu** [comment]
Apologies for the second reply, but I encountered a potential character issue. If a regex contains commas, using command line parameters will cause an unexpected behavior in which:

- The field name becomes the last string chunk following the last comma in the regex
- Validation fails since the regex is not valid anymore

This does not occur when using -jsonfile

Sample for comparison:

```
--textfield "Asset Tag",prompt="000000",regex='^\d{6}$',regexerror='Asset tag must be a six digit number.' \
--textfield "Computer Name",prompt="IST-CSS-ML-0000",regex='(^\\w{2,5}-(\\w{1,4}-)?[Mm]([Ll]|[Dd]|[Kk])-\\d{3,4}$)|(^(LNR|lnr|SPR|spr|TST|tst|RSV|rsv)-\\w{11,12}$)',regexError="Computer name does not align with CTS standard."
```

![regex-parsing-failed](https://user-images.githubusercontent.com/105297926/171183327-312a683b-9a99-4698-bf59-737f5c56b03e.png)

```
{
  "textfield" : [
    {
      "title" : "Asset Tag",
      "required" : true,
      "prompt" : "000000",
      "regex" : "^\\d{6}$",
      "regexerror" : "Asset tag must be a six digit number."
    },
    {
      "title" : "Computer Name",
      "required" : true,
      "prompt" : "IST-CSS-ML-0000",
      "regex" : "(^\\w{2,5}-(\\w{1,4}-)?[Mm]([Ll]|[Dd]|[Kk])-\\d{3,4}$)|(^(LNR|lnr|SPR|spr|TST|tst|RSV|rsv)-\\w{11,12}$)",
      "regexerror" : "Computer name does not align with CTS standard."
    }
  ]
}
```
![regex-parsing-ok](https://user-images.githubusercontent.com/105297926/171183697-679a58f1-2613-4e69-941d-0c96de1c0833.png)

**Comment by bartreardon** [comment]
hmm - ok, I can look at that. It will be because `,` is being used as the field separator in the argument so  will see `(^\w{2,5}` and break the regex command after the `2`. Will have a tinker and see what I can come up with.

**Comment by bartreardon** [comment]
json of course doesn't have that issue as it's json, not a string of text being interpreted as CSV

**Comment by dan-snelson** [comment]
+1

I learned today that **most** of our Asset Tags are seven-digits, so it'd be nice if I could get `^[0-9].{6,}$` to work.

**Comment by bartreardon** [comment]
ok - I think I have a neat way to handle this without requiting any syntax changes. Will post a preview build to the swiftdialog slack later today

**Comment by bartreardon** [comment]
available in [1.11](https://github.com/bartreardon/swiftDialog/releases/tag/v1.11.0)
